README.md

```diff
@@ -3,7 +3,71 @@
+- Baggage Context should be passed as: **the last parameter in the required non-function parameters group in a function declaration**.
+
+This way when reading the call side, users of these APIs can learn to "ignore" or "skim over" the context parameter and the method signature remains human-readable and “Swifty”.
+
+Examples: 
+
+- `func request(_ url: URL, context: BaggageContext)`, which may be called as `httpClient.request(url, context: context)`
+- `func handle(_ request: RequestObject,` **`context: FrameworkContext`** `)`
+  - if a "framework context" exists and _carries_ the baggage context already, it is permitted to pass that context together with the baggage;
+  - it is _strongly recommended_ to store the baggage context as `baggage` property of `FrameworkContext` in such cases, in order to avoid the confusing spelling of `context.context`, and favoring the self-explanatory `context.baggage` spelling when the baggage is contained in a framework context object.
+- `func receiveMessage(_ message: Message, context: FrameworkContext)`
+- `func handle(element: Element,` **`context: FrameworkContext`** `, settings: Settings? = nil)
+  - before any defaulted non-function parameters
+- `func handle(element: Element,` **`context: FrameworkContext`** `, settings: Settings? = nil, onComplete: () -> ())
+  - before defaulted parameters, which themselfes are before required function parameters
+- `func handle(element: Element,` **`context: FrameworkContext`** `, onError: (Error) -> (), onComplete: (() -> ())? = nil)
+
+In case there are _multiple_ "framework-ish" parameters, such as passing an NIO `EventLoop` or similar, we suggest:
+
+- `func perform(_ work: Work, for user: User,` _`frameworkThing: Thing, eventLoop: NIO.EventLoop,`_ **`context: BaggageContext`** `)`
+  - pass the baggage as **last** of such non-domain specific parameters as it will be _by far more_ omnipresent than any specific framework parameter - as it is expected that any framework should be accepting a context if it is able to do so. While not all libraries are necessarily going to be implemented using the same frameworks.
+
+And lastly
+
+We feel it is important to preserve Swift's human-readable nature of function definitions. In other words, we intend to keep the read-out-loud phrasing of methods to remain _"request that url (ignore reading out loud the context parameter)"_ rather than _"request (ignore this context parameter when reading) that url"_.
+
+
+#### Existing context argument
+
+When adapting an existing library/framework to support `BaggageContext` and it already has a "framework context" which is expected to be passed through "everywhere", we suggest to follow these guidelines to adopting BaggageContext:
+
+1. Add an `BaggageContext` as a property called `baggage` to your own `context` type, so that the call side for your users becomes `context.baggage` (rather than the confusing `context.context`)
```

Comment:
```suggestion
1. Add a `BaggageContext` as a property called `baggage` to your own `context` type, so that the call side for your users becomes `context.baggage` (rather than the confusing `context.context`)
```